t/24-worker-engine.t

```diff
@@ -227,13 +229,20 @@ subtest 'symlink testrepo' => sub {
     $result  = OpenQA::Worker::Engines::isotovideo::_link_repo($casedir, $pool_directory, 'opensuse');
     is $result, undef, 'create symlink successfully';
 
-    $settings->{DISTRI}   = 'mine';
-    $settings->{JOBTOKEN} = 'token99916';
-    delete $settings->{NEEDLES_DIR};
-    $settings->{CASEDIR} = 'https://github.com/foo/os-autoinst-distri-example.git#master';
-    $job                 = OpenQA::Worker::Job->new($worker, $client, {id => 12, settings => $settings});
-    $result              = OpenQA::Worker::Engines::isotovideo::engine_workit($job);
+    $settings->{DISTRI}      = 'fedora';
+    $settings->{JOBTOKEN}    = 'token99916';
+    $settings->{NEEDLES_DIR} = 'fedora/needles';
+    $settings->{CASEDIR}     = 'https://github.com/foo/os-autoinst-distri-example.git#master';
+    my $productdir = productdir('fedora', undef, undef);
+    $job = OpenQA::Worker::Job->new($worker, $client, {id => 12, settings => $settings});
+    combined_like { $result = OpenQA::Worker::Engines::isotovideo::engine_workit($job) }
+    qr /Symlinked from "$productdir\/needles" to "$pool_directory\/needles"/, 'symlink needles_dir';
     like $result->{child}->process_id, qr/\d+/, 'don\'t create symlink when CASEDIR is an url address';
+    my $vars_json = path($pool_directory)->child("vars.json")->slurp;
+    my $vars_data = decode_json $vars_json;
+    is $vars_data->{PRODUCTDIR}, $productdir,
+      'PRODUCTDIR is the default value when CASEDIR is a github address and not define PRODUCTDIR';
+    is $vars_data->{NEEDLES_DIR}, 'needles', 'When NEEDLES_DIR is a relative path, set it to basename';
```

Comment:
Can you confirm why you're changing the existing test? iirc that was working before. shouldn't we have two tests here?

Reply:
To be clear: the third commit, the new fix, also modified the existing test that *worked* before 🤔

Reply:
yes, but I see the modifications keep the original test expectations. In my eyes it's only *extended* which is ok.

Reply:
Previously, the test did `delete $settings->{NEEDLES_DIR};`.

Reply:
@kalikiana `deleting $settings->{NEEDLES_DIR} will not do symlink for needles_dir. I set another value in this test, it will do symlink, so the code will be covered. Just extending to cover more code.